cloudinit/sources/DataSourceGCE.py

```diff
@@ -29,14 +29,18 @@ class GoogleMetadataFetcher(object):
 
     def __init__(self, metadata_address):
         self.metadata_address = metadata_address
+        self.retries = 5
```

Comment:
It might be worth adding a url_sec_between_retries class attribute in the base Datasource class in cloudinit.sources.__init__. 

There are some common base class attributes like [url_retries, url_timeout etc](https://github.com/canonical/cloud-init/blob/main/cloudinit/sources/__init__.py#L177-L180) that are sourced by calling [self.get_url_params()](https://github.com/canonical/cloud-init/blob/main/cloudinit/sources/__init__.py#L391).   


This is helpful from a datasource config and testing perspective statically set them as class attributes url_retries url_timeout etc because they can also be overridden in with datasource configuration on disk via a file like /etc/cloud/cloud.cfg.d/customize-gcp.cfg
```yaml
datasource:
    GCE:
      retries: 10 # (defaults to 5 times)
      sec_between_retries: 10 # (defaults to 1 second)
```